include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -227,7 +227,7 @@ public :
                 && point_y >= geometry::get<min_corner, 1>(section.bounding_box) - 1
                 && point_y <= geometry::get<max_corner, 1>(section.bounding_box) + 1)
             {
-                for (int i = section.begin_index + 1; i <= section.end_index; i++)
+                for (signed_index_type i = section.begin_index + 1; i <= section.end_index; i++)
```

Comment:
section begin_index/end_index should be made signed_index_type too.
But that can be done separately, indeed.
So this signed_index_type here is OK